web/src/js/ducks/ui/flow.js

```diff
@@ -87,9 +88,31 @@ export function updateEdit(update) {
     return { type: UPDATE_EDIT, update }
 }
 
-export function stopEdit(flow) {
+export function uploadContent(flow, content, type){
     return (dispatch) => {
-        dispatch(flowsActions.update(flow, flow)).then(() => {
+            dispatch(flowsActions.updateContent(flow, content, type)).then( () => {
+            dispatch(flowsActions.updateFlow(flow))
+            dispatch({ type: STOP_EDIT })
+        })
+    }
+}
+
+export function stopEdit(modified_flow, old_flow) {
+    //make diff of modified_flow and old_flow
+    return (dispatch) => {
+        let flow = {...modified_flow}
+
+        if (flow.response.content) {
+            dispatch(flowsActions.updateContent(flow, flow.response.content, "response"))
+            flow.response = _.omit(flow.response, "content")
```

Comment:
If this is a copy anyway (L103), we  can just call `delete flow.response.content`.